ui/src/state.ts

```diff
@@ -0,0 +1,29 @@
+export class State {
+  constructor() {
+    this.included = new Array<Group>();
+  }
+
+  included: Array<Group>;
+}
+
+export class Group {
```

Comment:
It's possible you have a need for this to be a class for later, but an alternative that you can use with Typescript is instead (similar to a Go struct):

1. Just define these kinds of types as interfaces:

```
export interface Group {
  id: string
  criteria: Array<Criteria>
}
```

2. create instances via Javascript object literals

```
const g: Group = {
  id: 123,
  criteria: [...]
}
```

With this pattern I haven't had too much need for defining any custom model classes

Reply:
I already know I need to make major changes to what's here as part of the next steps so I'll keep approach in mind. Is the advantage just that the constructors are unnecessary or are there other reasons this is better?